src/System.Collections/src/System/Collections/Generic/List.cs

```diff
@@ -567,22 +567,19 @@ public int FindLastIndex(int startIndex, int count, Predicate<T> match)
                 // Special case for 0 length List
                 if (startIndex != -1)
                 {
-                    throw new ArgumentOutOfRangeException("startIndex", SR.ArgumentOutOfRange_Index);
+                    throw new ArgumentOutOfRangeException("startIndex", startIndex, SR.ArgumentOutOfRange_Index);
                 }
             }
-            else
+            // Make sure we're not out of range            
```

Comment:
Nit: we try not to have comments in a position like this.  It'd be fine at the end of the ```else if``` line